zerver/tests/test_decorators.py

```diff
@@ -2251,3 +2251,89 @@ def request_user_agent(user_agent: str) -> Tuple[Client, str]:
         # client_name has the full name still, though
         self.assertEqual(client_name, "very-long-name-goes-here-and-still-works")
         self.assert_length(queries, 0)
+
+
+class TestIgnoredParametersUnsupported(ZulipTestCase):
+    def test_ignored_parameters_json_success(self) -> None:
+        @has_request_variables
+        def test_view(
+            request: HttpRequest,
+            name: Optional[str] = REQ(default=None),
+            age: Optional[int] = 0,
+        ) -> HttpResponse:
+            return json_success(request)
+
+        # ignored parameter (not processed through REQ)
+        request = HostRequestMock()
+        request.POST["age"] = "30"
+        result = test_view(request)
+        self.assert_json_success(result, ["age"])
+
+        # valid parameter, returns no ignored parameters
+        request = HostRequestMock()
+        request.POST["name"] = "Hamlet"
+        result = test_view(request)
+        self.assert_json_success(result)
+
+        # both valid and ignored parameters
+        request = HostRequestMock()
+        request.POST["name"] = "Hamlet"
+        request.POST["age"] = "30"
+        request.POST["location"] = "Denmark"
+        request.POST["dies"] = "True"
+        result = test_view(request)
+        ignored_parameters = ["age", "location", "dies"]
```

Comment:
We should probably make the server sort the ignored parameters it returns; that's usually more readable. And then this would also be sorted.